File: src/main/java/com/syncstream_dev/syncstream/model/message/Message.java
package com.syncstream_dev.syncstream.model.message;

public abstract class Message {
    protected String messageId;
    protected long timestamp;
    protected String content;

    public Message(String messageId, long timestamp, String content) {
        this.messageId = messageId;
        this.timestamp = timestamp;
        this.content = content;
    }

    public String getMessageId() {
        return messageId;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public String getContent() {
        return content;
    }
}

File: syncstream/src/datamodules/message/message.rs
// message.rs implements the Message module

pub struct Message {
    id: Id,
    timestamp: Long,
    content: String,
}

impl Message {
    pub fn new(id: String, timestamp: Long, content: String) -> Message {
        Message {
            id,
            timestamp,
            content,
        }
    }
}
